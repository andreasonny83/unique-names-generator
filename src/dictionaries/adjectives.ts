export default [
  'able',
  'above',
  'absent',
  'absolute',
  'abstract',
  'abundant',
  'academic',
  'acceptable',
  'accepted',
  'accessible',
  'accurate',
  'accused',
  'active',
  'actual',
  'acute',
  'added',
  'additional',
  'adequate',
  'adjacent',
  'administrative',
  'adorable',
  'advanced',
  'adverse',
  'advisory',
  'aesthetic',
  'afraid',
  'aggregate',
  'aggressive',
  'agreeable',
  'agreed',
  'agricultural',
  'alert',
  'alive',
  'alleged',
  'allied',
  'alone',
  'alright',
  'alternative',
  'amateur',
  'amazing',
  'ambitious',
  'amused',
  'ancient',
  'angry',
  'annoyed',
  'annual',
  'anonymous',
  'anxious',
  'appalling',
  'apparent',
  'applicable',
  'appropriate',
  'arbitrary',
  'architectural',
  'arrogant',
  'artificial',
  'artistic',
  'ashamed',
  'asleep',
  'assistant',
  'associated',
  'atomic',
  'attractive',
  'automatic',
  'autonomous',
  'available',
  'average',
  'awake',
  'aware',
  'awful',
  'awkward',
  'back',
  'bad',
  'balanced',
  'basic',
  'beautiful',
  'beneficial',
  'better',
  'bewildered',
  'big',
  'binding',
  'biological',
  'bitter',
  'bizarre',
  'blank',
  'blonde',
  'blushing',
  'boiling',
  'bold',
  'bored',
  'boring',
  'bottom',
  'brainy',
  'brave',
  'breakable',
  'breezy',
  'brief',
  'bright',
  'brilliant',
  'broad',
  'broken',
  'bumpy',
  'burning',
  'busy',
  'calm',
  'capable',
  'careful',
  'casual',
  'causal',
  'cautious',
  'central',
  'certain',
  'changing',
  'characteristic',
  'charming',
  'cheap',
  'cheerful',
  'chemical',
  'chief',
  'chilly',
  'chosen',
  'chronic',
  'chubby',
  'circular',
  'civic',
  'civil',
  'civilian',
  'classic',
  'classical',
  'clean',
  'clear',
  'clever',
  'clinical',
  'close',
  'closed',
  'cloudy',
  'clumsy',
  'coastal',
  'cognitive',
  'coherent',
  'cold',
  'collective',
  'colorful',
  'colossal',
  'coloured',
  'colourful',
  'combative',
  'combined',
  'comfortable',
  'commercial',
  'common',
  'compact',
  'comparable',
  'comparative',
  'compatible',
  'competent',
  'competitive',
  'complete',
  'complex',
  'complicated',
  'comprehensive',
  'compulsory',
  'conceptual',
  'concerned',
  'concrete',
  'confident',
  'confidential',
  'confused',
  'conscious',
  'conservation',
  'conservative',
  'considerable',
  'consistent',
  'constant',
  'contemporary',
  'content',
  'continental',
  'continued',
  'continuing',
  'continuous',
  'controlled',
  'controversial',
  'convenient',
  'conventional',
  'convinced',
  'convincing',
  'cooing',
  'cool',
  'cooperative',
  'corporate',
  'correct',
  'corresponding',
  'costly',
  'courageous',
  'crazy',
  'creative',
  'creepy',
  'critical',
  'crooked',
  'crowded',
  'crucial',
  'crude',
  'cruel',
  'cuddly',
  'cultural',
  'curious',
  'curly',
  'current',
  'curved',
  'cute',
  'daily',
  'damaged',
  'damp',
  'dangerous',
  'dark',
  'dead',
  'deaf',
  'deafening',
  'dear',
  'decent',
  'decisive',
  'deep',
  'defeated',
  'defensive',
  'defiant',
  'definite',
  'deliberate',
  'delicate',
  'delicious',
  'delighted',
  'delightful',
  'democratic',
  'dependent',
  'depressed',
  'desirable',
  'desperate',
  'detailed',
  'determined',
  'developed',
  'developing',
  'devoted',
  'different',
  'difficult',
  'digital',
  'diplomatic',
  'direct',
  'dirty',
  'disabled',
  'disappointed',
  'disastrous',
  'disciplinary',
  'disgusted',
  'distant',
  'distinct',
  'distinctive',
  'distinguished',
  'disturbed',
  'disturbing',
  'diverse',
  'divine',
  'dizzy',
  'domestic',
  'dominant',
  'double',
  'doubtful',
  'drab',
  'dramatic',
  'dreadful',
  'driving',
  'dry',
  'dual',
  'due',
  'dull',
  'dusty',
  'dutch',
  'dying',
  'dynamic',
  'eager',
  'early',
  'eastern',
  'easy',
  'economic',
  'educational',
  'eerie',
  'effective',
  'efficient',
  'elaborate',
  'elated',
  'elderly',
  'eldest',
  'electoral',
  'electric',
  'electrical',
  'electronic',
  'elegant',
  'eligible',
  'embarrassed',
  'embarrassing',
  'emotional',
  'empirical',
  'empty',
  'enchanting',
  'encouraging',
  'endless',
  'energetic',
  'enormous',
  'enthusiastic',
  'entire',
  'entitled',
  'envious',
  'environmental',
  'equal',
  'equivalent',
  'essential',
  'established',
  'estimated',
  'ethical',
  'eventual',
  'everyday',
  'evident',
  'evil',
  'evolutionary',
  'exact',
  'excellent',
  'exceptional',
  'excess',
  'excessive',
  'excited',
  'exciting',
  'exclusive',
  'existing',
  'exotic',
  'expected',
  'expensive',
  'experienced',
  'experimental',
  'explicit',
  'extended',
  'extensive',
  'external',
  'extra',
  'extraordinary',
  'extreme',
  'exuberant',
  'faint',
  'fair',
  'faithful',
  'familiar',
  'famous',
  'fancy',
  'fantastic',
  'far',
  'fascinating',
  'fashionable',
  'fast',
  'fat',
  'favourable',
  'favourite',
  'federal',
  'fellow',
  'female',
  'feminist',
  'few',
  'fierce',
  'filthy',
  'final',
  'financial',
  'fine',
  'firm',
  'fiscal',
  'fit',
  'fixed',
  'flaky',
  'flat',
  'flexible',
  'fluffy',
  'fluttering',
  'flying',
  'following',
  'fond',
  'foolish',
  'foreign',
  'formal',
  'formidable',
  'forthcoming',
  'fortunate',
  'forward',
  'fragile',
  'frail',
  'frantic',
  'free',
  'frequent',
  'fresh',
  'friendly',
  'frightened',
  'front',
  'frozen',
  'full',
  'fun',
  'functional',
  'fundamental',
  'funny',
  'furious',
  'future',
  'fuzzy',
  'gastric',
  'general',
  'generous',
  'genetic',
  'gentle',
  'genuine',
  'geographical',
  'giant',
  'gigantic',
  'given',
  'glad',
  'glamorous',
  'gleaming',
  'global',
  'glorious',
  'golden',
  'good',
  'gorgeous',
  'gothic',
  'governing',
  'graceful',
  'gradual',
  'grand',
  'grateful',
  'greasy',
  'great',
  'grieving',
  'grim',
  'gross',
  'grotesque',
  'growing',
  'grubby',
  'grumpy',
  'guilty',
  'handsome',
  'happy',
  'hard',
  'harsh',
  'head',
  'healthy',
  'heavy',
  'helpful',
  'helpless',
  'hidden',
  'high',
  'hilarious',
  'hissing',
  'historic',
  'historical',
  'hollow',
  'holy',
  'homely',
  'hon',
  'honest',
  'horizontal',
  'horrible',
  'hostile',
  'hot',
  'huge',
  'human',
  'hungry',
  'hurt',
  'hushed',
  'husky',
  'icy',
  'ideal',
  'identical',
  'ideological',
  'ill',
  'illegal',
  'imaginative',
  'immediate',
  'immense',
  'imperial',
  'implicit',
  'important',
  'impossible',
  'impressed',
  'impressive',
  'improved',
  'inadequate',
  'inappropriate',
  'inc',
  'inclined',
  'increased',
  'increasing',
  'incredible',
  'independent',
  'indirect',
  'individual',
  'industrial',
  'inevitable',
  'influential',
  'informal',
  'inherent',
  'initial',
  'injured',
  'inland',
  'inner',
  'innocent',
  'innovative',
  'inquisitive',
  'instant',
  'institutional',
  'insufficient',
  'intact',
  'integral',
  'integrated',
  'intellectual',
  'intelligent',
  'intense',
  'intensive',
  'interested',
  'interesting',
  'interim',
  'interior',
  'intermediate',
  'internal',
  'international',
  'intimate',
  'invisible',
  'involved',
  'irrelevant',
  'isolated',
  'itchy',
  'jealous',
  'jittery',
  'joint',
  'jolly',
  'joyous',
  'judicial',
  'juicy',
  'junior',
  'just',
  'keen',
  'key',
  'kind',
  'known',
  'labour',
  'large',
  'late',
  'latin',
  'lazy',
  'leading',
  'left',
  'legal',
  'legislative',
  'legitimate',
  'lengthy',
  'lesser',
  'level',
  'lexical',
  'liable',
  'liberal',
  'light',
  'like',
  'likely',
  'limited',
  'linear',
  'linguistic',
  'liquid',
  'literary',
  'little',
  'live',
  'lively',
  'living',
  'local',
  'logical',
  'lonely',
  'long',
  'loose',
  'lost',
  'loud',
  'lovely',
  'low',
  'loyal',
  'ltd',
  'lucky',
  'mad',
  'magic',
  'magnetic',
  'magnificent',
  'main',
  'major',
  'male',
  'mammoth',
  'managerial',
  'managing',
  'manual',
  'many',
  'marginal',
  'marine',
  'marked',
  'married',
  'marvellous',
  'marxist',
  'mass',
  'massive',
  'mathematical',
  'mature',
  'maximum',
  'mean',
  'meaningful',
  'mechanical',
  'medical',
  'medieval',
  'melodic',
  'melted',
  'mental',
  'mere',
  'metropolitan',
  'mid',
  'middle',
  'mighty',
  'mild',
  'military',
  'miniature',
  'minimal',
  'minimum',
  'ministerial',
  'minor',
  'miserable',
  'misleading',
  'missing',
  'misty',
  'mixed',
  'moaning',
  'mobile',
  'moderate',
  'modern',
  'modest',
  'molecular',
  'monetary',
  'monthly',
  'moral',
  'motionless',
  'muddy',
  'multiple',
  'mushy',
  'musical',
  'mute',
  'mutual',
  'mysterious',
  'narrow',
  'nasty',
  'national',
  'native',
  'natural',
  'naughty',
  'naval',
  'near',
  'nearby',
  'neat',
  'necessary',
  'negative',
  'neighbouring',
  'nervous',
  'net',
  'neutral',
  'new',
  'nice',
  'noble',
  'noisy',
  'normal',
  'northern',
  'nosy',
  'notable',
  'novel',
  'nuclear',
  'numerous',
  'nursing',
  'nutritious',
  'nutty',
  'obedient',
  'objective',
  'obliged',
  'obnoxious',
  'obvious',
  'occasional',
  'occupational',
  'odd',
  'official',
  'ok',
  'okay',
  'old',
  'olympic',
  'only',
  'open',
  'operational',
  'opposite',
  'optimistic',
  'oral',
  'ordinary',
  'organic',
  'organisational',
  'original',
  'orthodox',
  'other',
  'outdoor',
  'outer',
  'outrageous',
  'outside',
  'outstanding',
  'overall',
  'overseas',
  'overwhelming',
  'painful',
  'pale',
  'panicky',
  'parallel',
  'parental',
  'parliamentary',
  'partial',
  'particular',
  'passing',
  'passive',
  'past',
  'patient',
  'payable',
  'peaceful',
  'peculiar',
  'perfect',
  'permanent',
  'persistent',
  'personal',
  'petite',
  'philosophical',
  'physical',
  'plain',
  'planned',
  'plastic',
  'pleasant',
  'pleased',
  'poised',
  'polite',
  'political',
  'poor',
  'popular',
  'positive',
  'possible',
  'potential',
  'powerful',
  'practical',
  'precious',
  'precise',
  'preferred',
  'preliminary',
  'premier',
  'prepared',
  'present',
  'presidential',
  'pretty',
  'previous',
  'prickly',
  'primary',
  'prime',
  'primitive',
  'principal',
  'printed',
  'prior',
  'private',
  'probable',
  'productive',
  'professional',
  'profitable',
  'profound',
  'progressive',
  'prominent',
  'promising',
  'proper',
  'proposed',
  'prospective',
  'protective',
  'protestant',
  'proud',
  'provincial',
  'psychiatric',
  'psychological',
  'public',
  'puny',
  'pure',
  'purring',
  'puzzled',
  'quaint',
  'qualified',
  'quarrelsome',
  'querulous',
  'quick',
  'quickest',
  'quiet',
  'quintessential',
  'quixotic',
  'radical',
  'rainy',
  'random',
  'rapid',
  'rare',
  'raspy',
  'rational',
  'ratty',
  'raw',
  'ready',
  'real',
  'realistic',
  'rear',
  'reasonable',
  'recent',
  'reduced',
  'redundant',
  'regional',
  'registered',
  'regular',
  'regulatory',
  'related',
  'relative',
  'relaxed',
  'relevant',
  'reliable',
  'relieved',
  'religious',
  'reluctant',
  'remaining',
  'remarkable',
  'remote',
  'renewed',
  'representative',
  'repulsive',
  'required',
  'resident',
  'residential',
  'resonant',
  'respectable',
  'respective',
  'responsible',
  'resulting',
  'retail',
  'retired',
  'revolutionary',
  'rich',
  'ridiculous',
  'right',
  'rigid',
  'ripe',
  'rising',
  'rival',
  'roasted',
  'robust',
  'rolling',
  'romantic',
  'rotten',
  'rough',
  'round',
  'royal',
  'rubber',
  'rude',
  'ruling',
  'running',
  'rural',
  'sacred',
  'sad',
  'safe',
  'salty',
  'satisfactory',
  'satisfied',
  'scared',
  'scary',
  'scattered',
  'scientific',
  'scornful',
  'scrawny',
  'screeching',
  'secondary',
  'secret',
  'secure',
  'select',
  'selected',
  'selective',
  'selfish',
  'semantic',
  'senior',
  'sensible',
  'sensitive',
  'separate',
  'serious',
  'severe',
  'shaggy',
  'shaky',
  'shallow',
  'shared',
  'sharp',
  'sheer',
  'shiny',
  'shivering',
  'shocked',
  'short',
  'shrill',
  'shy',
  'sick',
  'significant',
  'silent',
  'silky',
  'silly',
  'similar',
  'simple',
  'single',
  'skilled',
  'skinny',
  'sleepy',
  'slight',
  'slim',
  'slimy',
  'slippery',
  'slow',
  'small',
  'smart',
  'smiling',
  'smoggy',
  'smooth',
  'social',
  'socialist',
  'soft',
  'solar',
  'sole',
  'solid',
  'sophisticated',
  'sore',
  'sorry',
  'sound',
  'sour',
  'southern',
  'spare',
  'sparkling',
  'spatial',
  'special',
  'specific',
  'specified',
  'spectacular',
  'spicy',
  'spiritual',
  'splendid',
  'spontaneous',
  'sporting',
  'spotless',
  'spotty',
  'square',
  'squealing',
  'stable',
  'stale',
  'standard',
  'static',
  'statistical',
  'statutory',
  'steady',
  'steep',
  'sticky',
  'stiff',
  'still',
  'stingy',
  'stormy',
  'straight',
  'straightforward',
  'strange',
  'strategic',
  'strict',
  'striking',
  'striped',
  'strong',
  'structural',
  'stuck',
  'stupid',
  'subjective',
  'subsequent',
  'substantial',
  'subtle',
  'successful',
  'successive',
  'sudden',
  'sufficient',
  'suitable',
  'sunny',
  'super',
  'superb',
  'superior',
  'supporting',
  'supposed',
  'supreme',
  'sure',
  'surprised',
  'surprising',
  'surrounding',
  'surviving',
  'suspicious',
  'sweet',
  'swift',
  'symbolic',
  'sympathetic',
  'systematic',
  'tall',
  'tame',
  'tart',
  'tasteless',
  'tasty',
  'technical',
  'technological',
  'teenage',
  'temporary',
  'tender',
  'tense',
  'terrible',
  'territorial',
  'testy',
  'then',
  'theoretical',
  'thick',
  'thin',
  'thirsty',
  'thorough',
  'thoughtful',
  'thoughtless',
  'thundering',
  'tight',
  'tiny',
  'tired',
  'top',
  'tory',
  'total',
  'tough',
  'toxic',
  'traditional',
  'tragic',
  'tremendous',
  'tricky',
  'tropical',
  'troubled',
  'typical',
  'ugliest',
  'ugly',
  'ultimate',
  'unable',
  'unacceptable',
  'unaware',
  'uncertain',
  'unchanged',
  'uncomfortable',
  'unconscious',
  'underground',
  'underlying',
  'unemployed',
  'uneven',
  'unexpected',
  'unfair',
  'unfortunate',
  'unhappy',
  'uniform',
  'uninterested',
  'unique',
  'united',
  'universal',
  'unknown',
  'unlikely',
  'unnecessary',
  'unpleasant',
  'unsightly',
  'unusual',
  'unwilling',
  'upper',
  'upset',
  'uptight',
  'urban',
  'urgent',
  'used',
  'useful',
  'useless',
  'usual',
  'vague',
  'valid',
  'valuable',
  'variable',
  'varied',
  'various',
  'varying',
  'vast',
  'verbal',
  'vertical',
  'very',
  'vicarious',
  'vicious',
  'victorious',
  'violent',
  'visible',
  'visiting',
  'visual',
  'vital',
  'vitreous',
  'vivacious',
  'vivid',
  'vocal',
  'vocational',
  'voiceless',
  'voluminous',
  'voluntary',
  'vulnerable',
  'wandering',
  'warm',
  'wasteful',
  'watery',
  'weak',
  'wealthy',
  'weary',
  'wee',
  'weekly',
  'weird',
  'welcome',
  'well',
  'western',
  'wet',
  'whispering',
  'whole',
  'wicked',
  'wide',
  'widespread',
  'wild',
  'wilful',
  'willing',
  'willowy',
  'wily',
  'wise',
  'wispy',
  'wittering',
  'witty',
  'wonderful',
  'wooden',
  'working',
  'worldwide',
  'worried',
  'worrying',
  'worthwhile',
  'worthy',
  'written',
  'wrong',
  'xenacious',
  'xenial',
  'xenogeneic',
  'xeric',
  'xerothermic',
  'yabbering',
  'yammering',
  'yappiest',
  'yappy',
  'yawning',
  'yearling',
  'yearning',
  'yelling',
  'yelping',
  'yielding',
  'yodelling',
  'young',
  'youngest',
  'youthful',
  'ytterbic',
  'yucky',
  'yummy',
  'zany',
  'zealous',
  'zeroth',
  'zestful',
  'zesty',
  'zippy',
  'zonal',
  'zoophagous',
  'zygomorphic',
  'zygotic',
];
