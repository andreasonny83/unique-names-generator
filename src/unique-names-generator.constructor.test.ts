import { UniqueNamesGenerator, Config } from './unique-names-generator.constructor';
import { NumberDictionary } from './dictionaries';

describe('randomNameGenerator', () => {
  it('should exists', () => {
    // Arrange
    const config: Config = {
      dictionaries: [[], [], []],
      separator: '_',
      length: 3,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    // Assert
    expect(uniqueNamesGenerator).toBeDefined();
  });

  it('should have a generate method ', () => {
    // Arrange
    const config: Config = {
      dictionaries: [[], [], []],
      separator: '_',
      length: 3,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    // Assert
    expect(uniqueNamesGenerator.generate).toBeDefined();
  });

  it('generate: should return nothing', () => {
    // Arrange
    const config: Config = {
      dictionaries: [[], [], []],
      separator: '-',
      length: 3,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toEqual('');
  });

  it('generate: should return a string', () => {
    // Arrange
    const config: Config = {
      dictionaries: [[], [], []],
      separator: '_',
      length: 3,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(typeof response).toEqual('string');
  });

  it('generate: should generate a random name', () => {
    // Arrange
    const config: Config = {
      dictionaries: [['a'], ['b'], ['c']],
      separator: '-',
    };
    const expected = 'a-b-c';

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toEqual(expected);
  });

  it('generate: should generate a random name given only 2 dictionaries', () => {
    // Arrange
    const config: Config = {
      dictionaries: [['a'], ['b']],
      separator: '-',
    };
    const expected = 'a-b';

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toEqual(expected);
  });

  it('generate: should generate a random name given only 1 dictionary', () => {
    // Arrange
    const config: Config = {
      dictionaries: [['a']],
      separator: '-',
    };
    const expected = 'a';

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toEqual(expected);
  });

  it('generate: should accept a custom separator character', () => {
    // Arrange
    const config: Config = {
      dictionaries: [['a'], ['b'], ['c']],
      separator: '_',
      length: 3,
    };
    const expected = 'a_b_c';

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toEqual(expected);
  });

  it('should generate random combinations', () => {
    // Arrange
    const adjectives = ['Adjective1', 'Adjective2', 'Adjective3'];
    const colors = ['Color1', 'Color2', 'Color3'];
    const subjects = ['Animal1', 'Animal2', 'Animal3'];
    const config: Config = {
      dictionaries: [adjectives, colors, subjects],
      separator: '-',
      length: 3,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const response = uniqueNamesGenerator.generate();

    // Assert
    expect(response).toMatch(/^Adjective[123]-Color[123]-Animal[123]$/);
  });

  it('should generate only 2 word when depth is set to 2', () => {
    // Arrange
    const adjectives = ['Adjective1', 'Adjective2', 'Adjective3'];
    const colors = ['Color1', 'Color2', 'Color3'];
    const subjects = ['Animal1', 'Animal2', 'Animal3'];
    const config: Config = {
      dictionaries: [adjectives, subjects, colors],
      separator: '-',
      length: 2,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const expected = uniqueNamesGenerator.generate();

    // Assert
    expect(expected).toMatch(/^Adjective[123]-Animal[123]$/);
  });

  it('should generate the same name given an equal seed', () => {
    // Arrange
    const adjectives = ['Adjective1', 'Adjective2', 'Adjective3'];
    const colors = ['Color1', 'Color2', 'Color3'];
    const subjects = ['Animal1', 'Animal2', 'Animal3'];
    const config: Config = {
      dictionaries: [adjectives, subjects, colors],
      separator: '-',
      seed: 120498,
    };

    // Act
    let uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name1 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name2 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name3 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name4 = uniqueNamesGenerator.generate();

    // Assert
    expect(name1).toEqual(name2);
    expect(name1).toEqual(name3);
    expect(name1).toEqual(name4);
  });

  it('should generate the same name given an equal string seed', () => {
    // Arrange
    const adjectives = ['Adjective1', 'Adjective2', 'Adjective3'];
    const colors = ['Color1', 'Color2', 'Color3'];
    const subjects = ['Animal1', 'Animal2', 'Animal3'];
    const config: Config = {
      dictionaries: [adjectives, subjects, colors],
      separator: '-',
      seed: 'seed as a string',
    };

    // Act
    let uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name1 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name2 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name3 = uniqueNamesGenerator.generate();
    uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const name4 = uniqueNamesGenerator.generate();

    // Assert
    expect(name1).toEqual(name2);
    expect(name1).toEqual(name3);
    expect(name1).toEqual(name4);
  });

  it('should throw an error when there are no dictionaries', () => {
    // Arrange
    const config = {} as Config;

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const expected = (): unknown => uniqueNamesGenerator.generate();

    // Assert
    expect(() => expected()).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when there are no enough dictionaries', () => {
    // Arrange
    const config: Config = {
      dictionaries: [],
      length: 2,
      separator: '_',
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const expected = (): unknown => uniqueNamesGenerator.generate();

    // Assert
    expect(() => expected()).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when the provided length is invalid', () => {
    // Arrange
    const config: Config = {
      dictionaries: [],
      length: -1,
      separator: '_',
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
    const expected = (): unknown => uniqueNamesGenerator.generate();

    // Assert
    expect(() => expected()).toThrowErrorMatchingSnapshot();
  });

  describe('style', () => {
    it('should return a lower case formatted name when style is set to "lowerCase"', () => {
      // Arrange
      const config: Config = {
        dictionaries: [['TEST'], ['default'], ['style']],
        length: 3,
        separator: '_',
        style: 'lowerCase',
      };

      const expectedName = 'test_default_style';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(result).toEqual(expectedName);
    });

    it('should return a capitalized formatted name when style is set to "capital"', () => {
      // Arrange
      const config: Config = {
        dictionaries: [['test'], ['default'], ['style']],
        length: 3,
        separator: '_',
        style: 'capital',
      };

      const expectedName = 'Test_Default_Style';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(result).toEqual(expectedName);
    });

    it('should return an upper case formatted name when style is set to "upperCase"', () => {
      // Arrange
      const config: Config = {
        dictionaries: [['test'], ['default'], ['style']],
        length: 3,
        separator: '_',
        style: 'upperCase',
      };

      const expectedName = 'TEST_DEFAULT_STYLE';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(result).toEqual(expectedName);
    });

    it('should return an upper case formatted name even when some number dictionaries are present', () => {
      // Arrange
      const config: Config = {
        dictionaries: [['test'], ['default'], NumberDictionary.generate({ min: 1, max: 1 })],
        length: 3,
        separator: '_',
        style: 'upperCase',
      };

      const expectedName = 'TEST_DEFAULT_1';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(result).toEqual(expectedName);
    });

    it('should not throw any error when the word in the dictionary is empty and a formatted style is provided', () => {
      // Arrange
      const config: Config = {
        dictionaries: [[], [], []],
        length: 3,
        separator: '_',
        style: 'lowerCase',
      };

      const expectedName = '';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(result).toEqual(expectedName);
    });

    it('should use a number dictionary', () => {
      jest.spyOn(NumberDictionary, 'generate').mockReturnValue(['123']);

      // Arrange
      const numberDictionary = NumberDictionary.generate();
      const config: Config = {
        dictionaries: [['Dangerous'], ['Snake'], numberDictionary],
        length: 3,
        separator: '',
        style: 'capital',
      };

      const expectedName = 'DangerousSnake123';

      // Act
      const uniqueNamesGenerator = new UniqueNamesGenerator(config);
      const result = uniqueNamesGenerator.generate();

      // Assert
      expect(NumberDictionary.generate).toHaveBeenCalled();
      expect(result).toEqual(expectedName);
    });
  });
});
