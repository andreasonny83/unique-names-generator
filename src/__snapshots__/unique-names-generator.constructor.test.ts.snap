// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`randomNameGenerator should throw an error when the provided length is invalid 1`] = `"Invalid length provided"`;

exports[`randomNameGenerator should throw an error when there are no dictionaries 1`] = `"Cannot find any dictionary. Please provide at least one, or leave the \\"dictionary\\" field empty in the config object"`;

exports[`randomNameGenerator should throw an error when there are no enough dictionaries 1`] = `
"The length cannot be bigger than the number of dictionaries.
Length provided: 2. Number of dictionaries provided: 0"
`;
